packages/plugin-ext/src/main/browser/custom-editors/custom-editor-opener.tsx

```diff
@@ -68,14 +69,19 @@ export class CustomEditorOpener implements OpenHandler {
         if (widget?.isAttached) {
             return this.shell.activateWidget(widget.id);
         }
+        const stringUri = uri.toString();
+        if (!widget && this.pendingGetWidgets.has(stringUri)) {
+            return;
```

Comment:
This resolves to "undefined" if a widget is already being opened. Wouldn't it be better if we returned a promise that resolves to the widget that is already being opened? I mean something like this:

```suggestion
        if (!widget) {
            const uriString = uri.toString();
            let creation = this.pendingCreations.get(uriString);
            if (!creation) {
                const id = v4();
                creation = this.widgetManager.getOrCreateWidget<CustomEditorWidget>(CustomEditorWidget.FACTORY_ID, { id }).then(w => {
                    this.pendingCreations.delete(uriString);
                    w.viewType = this.editor.viewType;
                    w.resource = uri;
                    this.onDidOpenCustomEditorEmitter.fire(w);
                    return w;
                });
                this.pendingCreations.set(uriString, creation!);
            }
            return creation;
        }
```

Reply:
Much better, Thanks! 
Pushed the fix..

Reply:
I didn't test it though 🦆

Reply:
My fault, I didn't `yarn`... now it's working